devito/symbolics/extended_sympy.py

```diff
@@ -796,6 +794,51 @@ def __str__(self):
     __repr__ = __str__
 
 
+class VectorAccess(Expr, Pickable):
+
+    """
+    Represent a vector access operation at high-level.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        return Expr.__new__(cls, *args)
+
+    def __str__(self):
+        return f"VL4<{self.base}>"
+
+    __repr__ = __str__
+
+    func = Pickable._rebuild
+
+    def _sympystr(self, printer):
+        return str(self)
+
+    @property
+    def base(self):
+        return self.args[0]
+
+    @property
+    def function(self):
+        return self.base.function
+
+    @property
+    def indices(self):
+        return self.base.indices
+
+    @property
+    def dtype(self):
+        return self.function.dtype
+
+    @cacheit
+    def sort_key(self, order=None):
+        # Ensure that the VectorAccess is sorted as the base
+        return self.base.sort_key(order=order)
+
+    # Default assumptions correspond to those of the `base`
+    for i in ('is_real', 'is_imaginary', 'is_commutative'):
```

Comment:
these should all  be `_eval_is_real, ...`

Reply:
Yes. Even better, I'm now subclassing from BasicWrapperMixin, and apparently that's all we need. So I could drop 15~20 lines of code